src/main/java/it/pn/frontend/e2e/pages/destinatario/personaGiuridica/DeleghePGPagoPAPage.java

```diff
@@ -126,4 +139,104 @@ public boolean controlloDelegaRestituita(String ragioneSociale) {
+    }
+
+    public void clickBottoneConferma() {
+        this.confermaButton.click();
+    }
+
+
+    public void controlloStatoAttiva(String ragioneSociale) {
+        try {
+            By statoAttivaBy = By.xpath("//tr[@data-testid='table(notifications).row']//td[@role='cell' and div/p[contains(text(),'"+ragioneSociale+"')]]/following-sibling::td[@role='cell']//div/div/span[contains(text(),'Attiva')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(statoAttivaBy));
+            logger.info("La delega ha lo stato Attiva");
+        }catch (TimeoutException e){
+            logger.error("La delega NON ha lo stato Attiva con errore: "+e.getMessage());
+            Assert.fail("La delega NON ha lo stato Attiva con errore: "+e.getMessage());
+        }
+    }
+
+    public void clickNonAssegnaGruppo() {
+        this.nonGruppoRadioButton.click();
+    }
+
+    public void clickOpzioneRifiuta() {
+        this.opzioneRevoca.click();
+    }
+
+    public void clickBottoneRifiuta() {
+        this.rifiutaButton.click();
+    }
+
+    public void waitLoadPopUpRevoca() {
+        try {
+            By revocaPopUpBy = By.xpath("//div[@aria-labelledby='responsive-dialog-title']");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(revocaPopUpBy));
+            this.logger.info("Si visualizza il pop-up rifiuta delega");
+        }catch (TimeoutException e){
+            this.logger.error("Non si visualizza il pop-up rifiuta delega con errore: "+e.getMessage());
+            Assert.fail("Non si visualizza il pop-up rifiuta delega con errore: "+e.getMessage());
+        }
+    }
+
+    public void clickOpzioneModifica() {
+        this.opzioneModifica.click();
+    }
+
+    public void waitLoadPopUpModifica() {
+        try {
+            By titlePOPUPBy = By.id("dialog-title");
+            By nonAssegnaButtonBy = By.xpath("//span[@data-testid='no-group']");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(titlePOPUPBy));
+            this.getWebDriverWait(30).until(ExpectedConditions.elementToBeClickable(nonAssegnaButtonBy));
+            logger.info("Si visualizza correttamente il pop-up");
+        }catch (TimeoutException e){
+            logger.error("Si visualizza NON correttamente il pop-up con errore: "+e.getMessage());
+            Assert.fail("Si visualizza NON correttamente il pop-up con errore: "+e.getMessage());
+        }
+    }
+
+    public boolean verificaPresenzaGruppo(String ragioneSociale) {
+        try {
+            By gruppoBy = By.xpath("//td[@role='cell' and div/p[contains(text(),'"+ragioneSociale+"')]]/following-sibling::td[@role='cell']//span[contains(text(),'Test gruppi')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(gruppoBy));
+            return true;
+        }catch (TimeoutException e) {
+            return false;
```

Comment:
insert log

Reply:
Logger already present in the step that calls this method